src/Analyzers/MSTest.Analyzers/DoNotUseSystemDescriptionAttributeAnalyzer.cs

```diff
@@ -0,0 +1,80 @@
+    private static readonly LocalizableResourceString Description = new(nameof(Resources.DoNotUseSystemDescriptionAttributeDescription), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString MessageFormat = new(nameof(Resources.DoNotUseSystemDescriptionAttributeMessageFormat), Resources.ResourceManager, typeof(Resources));
+
+    internal static readonly DiagnosticDescriptor DoNotUseSystemDescriptionAttributeRule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.DoNotUseSystemDescriptionAttributeRuleId,
+        Title,
+        MessageFormat,
+        Description,
+        Category.Usage,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+        = ImmutableArray.Create(DoNotUseSystemDescriptionAttributeRule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestMethodAttribute, out INamedTypeSymbol? testMethodAttributeSymbol)
+                && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDescriptionAttribute, out INamedTypeSymbol? systemDescriptionAttributeSymbol))
+            {
+                context.RegisterSymbolAction(
+                    context => AnalyzeSymbol(context, testMethodAttributeSymbol, systemDescriptionAttributeSymbol),
+                    SymbolKind.Method);
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol testMethodAttributeSymbol, INamedTypeSymbol systemDescriptionAttributeSymbol)
+    {
+        var methodSymbol = (IMethodSymbol)context.Symbol;
+
+        bool hasTestMethodAttribute = false;
+        bool hasSystemDescriptionAttribute = false;
+        foreach (AttributeData attribute in methodSymbol.GetAttributes())
+        {
+            if (attribute.AttributeClass.Inherits(testMethodAttributeSymbol))
+            {
+                hasTestMethodAttribute = true;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, systemDescriptionAttributeSymbol))
+            {
+                hasSystemDescriptionAttribute = true;
+            }
+
+            if (hasTestMethodAttribute && hasSystemDescriptionAttribute)
+            {
+                break;
+            }
+        }
+
+        if (hasTestMethodAttribute && hasSystemDescriptionAttribute)
+        {
+            context.ReportDiagnostic(methodSymbol.CreateDiagnostic(DoNotUseSystemDescriptionAttributeRule, methodSymbol.Name));
+        }
```

Comment:
It's good as-is, you could also have reported and break:

```suggestion
            if (hasTestMethodAttribute && hasSystemDescriptionAttribute)
            {
                context.ReportDiagnostic(methodSymbol.CreateDiagnostic(DoNotUseSystemDescriptionAttributeRule, methodSymbol.Name));
                break;
            }
        }
```